### PR TITLE
Middlewares for the win!

### PR DESCRIPTION
**Pull request by dgaubert**
- Extract query params parse to its middleware

### CODE REVIEW

File: test/acceptance/log.js [hunk 1]
```diff
@@ -24,34 +24,6 @@ describe('Log middleware', function() {
 
     describe('regular queries endpoint', function() {
         ['GET', 'POST'].forEach(method => {
-            it(`${method} without query fails`, function(done) {
```
[the remark this answers is not in the document]
Comment (dgaubert) [reply test/acceptance/log.js]: Does have any sense log  `null` when the body of the response is `sql param is required`?

File: app/middlewares/parameters.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
+'use strict';
+
+const sanitizeFilename = require('../utils/filename_sanitizer');
+const formats = require('../models/formats');
+
+module.exports = function parameters ({ strategy = 'query' } = {}) {
+    const getParameters = getParametersFromStrategy(strategy);
+
+    return function parametersMiddleware (req, res, next) {
+        const input = Object.assign({}, req.query, req.body || {});
```
[the remark this answers is not in the document]
Comment (dgaubert) [reply app/middlewares/parameters.js]: handled here: https://github.com/CartoDB/CartoDB-SQL-API/pull/607/files#diff-66de868447d8d6709a250b22d61798deR75

File: test/acceptance/log.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dgaubert) [reply test/acceptance/log.js]: These corner case makes to change the order of the middlewares `log`, `params` depending on the strategy `query`, `job`, `copy`

[the remark this answers is not in the document]
Comment (oleurud) [reply test/acceptance/log.js]: Ok, understood

File: app/middlewares/parameters.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (oleurud) [reply app/middlewares/parameters.js]: Ah, right! :+1:
